Training/Training/Change.c

```diff
@@ -0,0 +1,40 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Change.c
+// Program on main branch.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+
+void ChangeRequired (int* balance, int* change10, int* change5, int* change2, int* change1) {
+   int num = *balance;
+   if (num >= 10) {
+      *change10 = num / 10;
+      num %= 10;
+   }
+   if (num >= 5) {
+      *change5 = num / 5;
+      num %= 5;
+   }
+   if (num >= 2) {
+      *change2 = num / 2;
+      num %= 2;
+   }
+   if (num >= 1) {
+      *change1 = num / 1;
+      num %= 1;
+   }
+}
+
+void main () {
+   int balances[] = { 5,18,1,43,500 }, expResult[][4] = { { 0,1,0,0 },{ 1,1,1,1 },{ 0,0,0,1 },{ 4,0,1,1 },{ 50,0,0,0} };
+   printf ("TestCases:\n");
+   for (int i = 0; i < 5; i++) {
```

Comment:
The loop terminator 5 is a "magic" number. If you change the number of tests, you have to remember to change this also. Further, in more complicated programs, if someone has to modify your code, they will have no idea what 5 means. You can replace it with a variable which is automatically calculated by the size of the array "balances".